plugins/github-actions/src/api/GithubActionsClient.ts

```diff
@@ -15,117 +15,94 @@
  */
 
 import { GithubActionsApi } from './GithubActionsApi';
-import { Build, BuildDetails, BuildStatus, WorkflowRun } from './types';
-
-const statusToBuildStatus: { [status: string]: BuildStatus } = {
-  success: BuildStatus.Success,
-  failure: BuildStatus.Failure,
-  pending: BuildStatus.Pending,
-  running: BuildStatus.Running,
-  in_progress: BuildStatus.Running,
-  completed: BuildStatus.Success,
-};
-
-const conclusionToStatus = (conslusion: string): BuildStatus =>
-  statusToBuildStatus[conslusion] ?? BuildStatus.Null;
+import { Octokit } from '@octokit/rest';
+import {
+  ActionsListWorkflowRunsForRepoResponseData,
+  ActionsGetWorkflowResponseData,
+  ActionsGetWorkflowRunResponseData,
+} from '@octokit/types';
+
+// const statusToBuildStatus: { [status: string]: BuildStatus } = {
+//   success: BuildStatus.Success,
+//   failure: BuildStatus.Failure,
+//   pending: BuildStatus.Pending,
+//   running: BuildStatus.Running,
+//   in_progress: BuildStatus.Running,
+//   completed: BuildStatus.Success,
+// };
+
+// const conclusionToStatus = (conslusion: string): BuildStatus =>
+//   statusToBuildStatus[conslusion] ?? BuildStatus.Null;
 
 export class GithubActionsClient implements GithubActionsApi {
-  async listBuilds({
+  private api: Octokit;
+  constructor({ api }: { api: Octokit }) {
+    this.api = api;
+  }
+  reRunWorkflow({
     owner,
     repo,
-    token,
+    runId,
   }: {
     owner: string;
     repo: string;
-    token: string;
-  }): Promise<Build[]> {
-    const url = `https://api.github.com/repos/${owner}/${repo}/actions/runs`;
-
-    const response = await fetch(url, {
-      headers: new Headers({
-        Authorization: `Bearer ${token}`,
-      }),
+    runId: number;
+  }) {
+    this.api.actions.reRunWorkflow({
+      owner,
+      repo,
+      run_id: runId,
     });
-
-    if (!response.ok) {
-      return [
-        {
-          commitId: 'Error',
-          message: 'Response status is not OK',
-          branch: 'Error',
-          status: BuildStatus.Failure,
-          uri: 'Error',
-        },
-      ];
-    }
-
-    const data = await response.json();
-
-    const newData: WorkflowRun[] = data.workflow_runs;
-
-    const endData: Build[] = [];
-
-    newData.forEach((element, index) => {
-      const transData: Build = {
-        commitId: '',
-        message: '',
-        branch: '',
-        status: BuildStatus.Null,
-        uri: '',
-      };
-      transData.commitId = String(element.head_commit.id);
-      transData.branch = element.head_branch;
-      transData.status = conclusionToStatus(element.conclusion);
-      transData.message = element.head_commit.message;
-      transData.uri = element.url;
-      endData[index] = transData;
+  }
+  async listWorkflowRuns({
+    owner,
+    repo,
+    pageSize = 100,
+    page = 0,
```

Comment:
We don't have `token` here anymore, how does it work with auth then?

Reply:
I accidentally removed it. Will fix.